src/cls/rbd/cls_rbd.cc

```diff
@@ -711,6 +717,98 @@ int set_stripe_unit_count(cls_method_context_t hctx, bufferlist *in, bufferlist
   return 0;
 }
 
+/**
+ * get the image flags
+ *
+ * Input:
+ * @param snap_id which snapshot to query, to CEPH_NOSNAP (uint64_t)
```

Comment:
it seems like this would be better for transient state - which an invalid object map can be - so there's no need to store snapshots of it. it could be stored as part of the objectmap object instead